src/components/PlaybackControls/index.tsx

```diff
@@ -35,11 +35,36 @@ const PlayBackControls = ({
     timeStep,
     isEmpty,
 }: PlayBackProps) => {
+    const [unitIndex, setUnitIndex] = useState(0);
+    // Where to resume playing if simulation was playing before scrubbing
+    const [targetPlayTime, setTargetPlayTime] = useState(-1);
+
+    // - Gets called once when the user clicks on the slider to skip to a specific time
+    // - Gets called multiple times when user is scrubbing (every time the play head
+    //     passes through a time value associated with a frame)
     const handleTimeChange = (sliderValue: number | [number, number]): void => {
-        onTimeChange(sliderValue as number); // slider can be a list of numbers, but we're just using a single value
+        // sliderValue can be an array of numbers (representing a selected range),
+        // but we're just using a single value
+        onTimeChange(sliderValue as number);
+        if (isPlaying) {
+            // Need to save the sliderValue as targetPlayTime to use in handleSliderMouseUp,
+            // because the sliderValue argument available in handleSliderMouseUp is not accurate
+            // when the time between mouse down and mouse up is short.
+            setTargetPlayTime(sliderValue as number);
+            pauseHandler();
+        } else if (targetPlayTime >= 0) {
+            // Update targetPlayTime if user is still dragging
+            setTargetPlayTime(sliderValue as number);
+        }
     };
 
-    const [unitIndex, setUnitIndex] = useState(0);
+    const handleSliderMouseUp = (): void => {
+        // Resume playing if simulation was playing before
+        if (targetPlayTime >= 0) {
```

Comment:
to see if i understand, you are using -1 to indicate that there is "no target play time"? and that is because all valid times are always non-negative?

Reply:
yes

Reply:
Is targetPlayTime used anywhere else?  If not, then I would consider naming it something like savedPlayTimeWhileScrubbing to make it really obvious.

Reply:
haha, actually its original name was `wasPlayingBeforeScrubbing` (it started out as a boolean)... how about `timeToResumeAfterScrubbing`? @meganrm any opposition?

Reply:
sounds good to me.  targetPlayTime sounds like something that has a bigger meaning

Reply:
sounds good to me too

Reply:
Excellent I have renamed it